examples/imagenet/imagenet_lib_test.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Lint as: python3
+"""Tests for flax.examples.imagenet.imagenet_lib."""
+
+import itertools
+import os
+import tempfile
+
+from absl.testing import absltest
+
+from tensorboard.backend.event_processing import directory_watcher
+from tensorboard.backend.event_processing import event_file_loader
+from tensorboard.util import tensor_util
+import tensorflow_datasets as tfds
+
+import imagenet_lib
+
+
+# TODO(#290): Refactor logic in testing/benchmark.py to create a 
+# utility class for parsing event files and extracting scalar summaries. 
+def _process_event(event):
+  for value in event.summary.value:
+    yield value
+
+
+def _parse_and_return_summary_values(path):
+  """Parses event file in given `path` and returns scalar summaries logged."""
+  tag_event_value_dict = {}
+  event_file_generator = directory_watcher.DirectoryWatcher(
```

Comment:
Why do we need this? Are we really watching the directory? Isn't there a more straightforward API to just read the latest observations? (What did Alexey do in his original Benchmark PR?)

Reply:
This is required to test the recorded metrics as an assertion that the model training indeed happened.
So this code is a motivation from that PR. This is mostly copy pasted from there and I have #290 to address this duplication.

Reply:
Removed the assertions altogether.